confgenerator/testdata/valid/linux/logging-receiver_tomcat/golden_fluent_bit_main.conf

```diff
@@ -0,0 +1,194 @@
+    storage.max_chunks_up     128
+    storage.metrics           on
+    storage.sync              normal
+
+[INPUT]
+    Buffer_Chunk_Size 512k
+    Buffer_Max_Size   5M
+    DB                ${buffers_dir}/default_pipeline_syslog
+    Key               message
+    Mem_Buf_Limit     10M
+    Name              tail
+    Path              /var/log/messages,/var/log/syslog
+    Read_from_Head    True
+    Rotate_Wait       30
+    Skip_Long_Lines   On
+    Tag               default_pipeline.syslog
+    storage.type      filesystem
+
+[INPUT]
+    Buffer_Chunk_Size 512k
+    Buffer_Max_Size   5M
+    DB                ${buffers_dir}/tomcat_tomcat_access
+    Key               message
+    Mem_Buf_Limit     10M
+    Name              tail
+    Path              /opt/tomcat/logs/localhost_access_log.*.txt
+    Read_from_Head    True
+    Rotate_Wait       30
+    Skip_Long_Lines   On
+    Tag               tomcat.tomcat_access
+    storage.type      filesystem
+
+[INPUT]
+    Buffer_Chunk_Size 512k
+    Buffer_Max_Size   5M
+    DB                ${buffers_dir}/tomcat_tomcat_system
+    Key               message
+    Mem_Buf_Limit     10M
+    Name              tail
+    Path              /opt/tomcat/logs/catalina.out
+    Read_from_Head    True
+    Rotate_Wait       30
+    Skip_Long_Lines   On
+    Tag               tomcat.tomcat_system
+    storage.type      filesystem
+
+[INPUT]
+    Buffer_Chunk_Size 512k
+    Buffer_Max_Size   5M
+    DB                ${buffers_dir}/ops-agent-fluent-bit
+    Key               message
+    Mem_Buf_Limit     10M
+    Name              tail
+    Path              ${logs_dir}/logging-module.log
+    Read_from_Head    True
+    Rotate_Wait       30
+    Skip_Long_Lines   On
+    Tag               ops-agent-fluent-bit
+    storage.type      filesystem
+
+[FILTER]
+    Add   logging.googleapis.com/logName syslog
+    Match default_pipeline.syslog
+    Name  modify
+
+[FILTER]
+    Key_Name message
+    Match    tomcat.tomcat_access
+    Name     parser
+    Parser   tomcat.tomcat_access.tomcat_access
+
+[FILTER]
+    Condition Key_Value_Equals host -
+    Match     tomcat.tomcat_access
+    Name      modify
+    Remove    host
+
+[FILTER]
+    Condition Key_Value_Equals user -
+    Match     tomcat.tomcat_access
+    Name      modify
+    Remove    user
+
+[FILTER]
+    Condition Key_Value_Equals http_request_referer -
+    Match     tomcat.tomcat_access
+    Name      modify
+    Remove    http_request_referer
+
+[FILTER]
+    Match         tomcat.tomcat_access
+    Name          nest
+    Nest_under    logging.googleapis.com/http_request
+    Operation     nest
+    Remove_prefix http_request_
+    Wildcard      http_request_*
+
+[FILTER]
+    Add   logging.googleapis.com/logName tomcat_access
+    Match tomcat.tomcat_access
+    Name  modify
+
+[FILTER]
+    Match                 tomcat.tomcat_system
+    Multiline.Key_Content message
+    Multiline.Parser      tomcat.tomcat_system.tomcat_system.multiline
```

Comment:
Using multiline parser filter is unsafe. When the logs spans across two chunks, it can't concatenate them right. 
![image](https://user-images.githubusercontent.com/8354694/152434132-03f0b509-85f3-4555-a23f-ad6da8d367d3.png)
I tested it with systemd with below configuration, logs across two chunks didn't concatenate right:
```
sophieyfang_google_com@debian10-meow:~$ cat parsers_systemd.conf 
[PARSER]
    Name        parse_json_0
    Format      json
[MULTILINE_PARSER]
    name          multiline-regex-test
    key_content   MESSAGE
    type          regex
    flush_timeout 1000
    #
    # Regex rules for multiline parsing
    # ---------------------------------
    #
    # configuration hints:
    #
    #  - first state always has the name: start_state
    #  - every field in the rule must be inside double quotes
    #
    # rules |   state name  | regex pattern                  | next state
    # ------|---------------|--------------------------------------------
    rule      "start_state"   "(\d+-\d+-\d+T\d+:\d+:\d+.\d+Z)(.*)"  "cont"
    rule      "cont"          "go.opentelemetry.io.*"               "cont1"
    rule      "cont1"         "^\s+.*"                              "cont"
    
 
[OUTPUT]
    name                  stdout
    match                 *
    
sophieyfang_google_com@debian10-meow:~$ cat fluent-bit-json-systemd.conf 
[SERVICE]
    flush                 1
    log_level             info
    parsers_file          parsers_systemd.conf
 
[INPUT]
    name                  systemd
    Tag			  MEOW
    Max_Entries           15000
    Systemd_Filter        _SYSTEMD_UNIT=google-cloud-ops-agent-opentelemetry-collector.service
 
[FILTER]
    name                  multiline
    match                 *
    multiline.parser      go,multiline-regex-test
 
[OUTPUT]
    name                  stdout
    match                 *
```

Reply:
I think it's fine for 3rd party integration use multiline filter as there is no other options. But for ops-agent native support, we decide to only support [tail]

Reply:
Yeah, the current comprimise is:

* `tail` input plugin based 3rd party application logs => Use the multiline parser if possible
*  `syslog` input plugin based 3rd party application logs => Use the multiline filter as there is not parser available